docker-bake.hcl

```diff
@@ -32,6 +32,20 @@ function "parse_refs" {
   ]
 }
 
+function "ref_info" {
+  params = [ref]
+  result = {
+    cache_tag = (
+      ref == "v0.12.0" || ref == "18fc875d9bfd6e065cd8211abc639434ba65aa56" ? "v0.12.0-pick-pr-4361" :
+      ref
+    ),
+    context = (
+      ref == "v0.12.0" || ref == "18fc875d9bfd6e065cd8211abc639434ba65aa56" ? "https://github.com/crazy-max/buildkit.git#v0.12.0-pick-pr-4361" :
+      "https://github.com/${BUILDKIT_REPO}.git#${ref}"
+    )
+  }
+}
```

Comment:
@tonistiigi For patches I use this hcl func for now but ideally I would need to use a dockerfile inline to just build buildctl/buildkitd with a set of patches. This would probably be an extra build along `buildkit-build` target as we need vanilla packages for daemon and package size metrics.